plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -283,7 +285,20 @@ private function getSegmentIfIsUnprocessed()
             $storedSegment = null;
         }
 
-        return [$segment, $storedSegment, $isSegmentToPreprocess];
+
+        $lastArchiveData =  ArchiveInvalidator::getEarliestDateToRearchive();
+
+        $endDate = $period->getDateEnd();
+        $archived = true;
```

Comment:
Wouldn't a variable name like `$canBeArchived` make the code easier to understand?